clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java

```diff
@@ -166,9 +166,12 @@ public static GatewayStub buildGatewayStub(
       final ManagedChannel channel, final ZeebeClientConfiguration config) {
     final CallCredentials credentials = buildCallCredentials(config);
     final GatewayStub gatewayStub = GatewayGrpc.newStub(channel).withCallCredentials(credentials);
-    if (!config.getInterceptors().isEmpty()) {
-      return gatewayStub.withInterceptors(
-          config.getInterceptors().toArray(new ClientInterceptor[] {}));
+    if (config instanceof ZeebeClientBuilderImpl) {
```

Comment:
Why not pass the interceptors when constructing `ZeebeClientImpl` instead of casting here the config?